<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>singscore • singscore</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="singscore">
<meta property="og:description" content="A simple single-sample gene signature scoring method that uses rank-based 
  statistics to analyze the sample's gene expression profile. It scores the expression activities of 
  gene sets at a single-sample level.">
<meta property="og:image" content="https://DavisLaboratory.github.io/singscore/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">singscore</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/singscore.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/singscore" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="singscore-">singscore <img src="https://github.com/Bioconductor/BiocStickers/blob/master/singscore/singscore.png?raw=true" alt="logo" align="right" height="140" width="120"><a class="anchor" aria-label="anchor" href="#singscore-"></a>
</h1></div>

<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>‘singscore’ is an R/Bioconductor package which implements the simple single-sample gene-set (or gene-signature) scoring method proposed by Foroutan <em>et al.</em> (2018) and Bhuva <em>et al.</em> (2020). It uses rank-based statistics to analyze each sample’s gene expression profile and scores the expression activities of gene sets at a single-sample level.</p>
<p>Additional packages we have developed can enhance the singscore workflow:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://www.bioconductor.org/packages/release/data/experiment/html/msigdb.html" class="external-link"><code>msigdb</code></a> - A package that provides gene-sets from the molecular signatures database (MSigDB) as a <code>GeneSetCollection</code> object that is compatible with <code>singscore</code>.</li>
<li>
<a href="https://www.bioconductor.org/packages/release/bioc/html/vissE.html" class="external-link"><code>vissE</code></a> - A package that can summarise and aid in the interpretation of a list of significant gene-sets identified by <code>singscore</code> (see <a href="https://davislaboratory.github.io/GenesetAnalysisWorkflow/" class="external-link">tutorial</a>).</li>
<li>
<a href="https://www.bioconductor.org/packages/release/data/experiment/html/emtdata.html" class="external-link"><code>emtdata</code></a> - The full EMT dataset used in this tutorial (with additional EMT related datasets).</li>
</ol>
<p>We have also published and made openly available the extensive tutorials below that demonstrate the variety of ways in which <code>singscore</code> can be used to gain a better functional understanding of molecular data:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://f1000research.com/articles/8-776" class="external-link">Using singscore to predict mutation status in acute myeloid leukemia from transcriptomic signatures</a>.</li>
<li>
<a href="https://davislaboratory.github.io/GenesetAnalysisWorkflow/" class="external-link">Gene-set enrichment analysis workshop</a> - available through the <a href="http://app.orchestra.cancerdatasci.org/" class="external-link">Orchestra</a> platform (search “WEHI Masterclass Day 4: Functional Analysis, single sample gene set analysis”).</li>
</ol>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>These instructions will get you to install the package up and running on your local machine. If you experience any issues, please raise a GitHub issue at <a href="https://github.com/DavisLaboratory/singscore/issues" class="external-link uri">https://github.com/DavisLaboratory/singscore/issues</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># build_vignettes = TRUE to build vignettes upon installation</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"singscore"</span>, <span class="at">version =</span> <span class="st">"3.8"</span>)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>The package comes with a vignette showing how the different functions in the package can be used to perform a gene-set enrichment analysis on a single sample level. Pre-built vignettes can be accessed via <a href="https://bioconductor.org/packages/release/bioc/vignettes/singscore/inst/doc/singscore.html" class="external-link">Bioconductor</a> or <a href="https://davislaboratory.github.io/singscore/articles/singscore.html" class="external-link">the GitHub IO page</a>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Foroutan M, Bhuva D, Lyu R, Horan K, Cursons J, Davis M (2018). “Single sample scoring of molecular phenotypes.” <em>BMC bioinformatics</em>, <em>19</em>(1), 404. doi: <a href="https://doi.org/10.1186/s12859-018-2435-4" class="external-link">10.1186/s12859-018-2435-4</a>.</p>
<p>Bhuva D, Cursons J, Davis M (2020). “Stable gene expression for normalisation and single-sample scoring.” <em>Nucleic Acids Research</em>, <em>48</em>(19), e113. doi: <a href="https://doi.org/10.1093/nar/gkaa802" class="external-link">10.1093/nar/gkaa802</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/singscore" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/DavisLaboratory/singscore/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/DavisLaboratory/singscore/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing singscore</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ruqian Lyu <br><small class="roles"> Author, contributor </small>  </li>
<li>Momeneh Foroutan <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0002-1440-0457" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Dharmesh D. Bhuva <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6398-9157" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Malvika Kharbanda <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-9726-3023" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/DavisLaboratory/singscore/actions" class="external-link"><img src="https://github.com/DavisLaboratory/singscore/workflows/R-CMD-check-bioc/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://codecov.io/gh/DavisLaboratory/singscore" class="external-link"><img src="https://codecov.io/gh/DavisLaboratory/singscore/branch/master/graph/badge.svg?token=OWOL51QJD1" alt="codecov"></a></li>
<li><a href="https://bioconductor.org/packages/singscore/" class="external-link"><img src="https://bioconductor.org/shields/years-in-bioc/singscore.svg" alt="BioC status"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Ruqian Lyu, Momeneh Foroutan, Dharmesh D. Bhuva, Malvika Kharbanda.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
