<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>singscore • singscore</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="singscore">
<meta property="og:description" content="singscore">
<meta property="og:image" content="https://DavisLaboratory.github.io/singscore/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singscore</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.19.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/singscore.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/singscore" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>singscore</h1>
                        <h4 data-toc-skip class="author">Ruqian Lyu</h4>
            
            <h4 data-toc-skip class="date">2023-06-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/singscore/blob/HEAD/vignettes/singscore.Rmd" class="external-link"><code>vignettes/singscore.Rmd</code></a></small>
      <div class="hidden name"><code>singscore.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use the functions in R package
‘singscore’ to score a gene expression dataset against a gene set at a
single-sample level and provides visualisation functions to improve
interpretation of the results.</p>
<p>Please cite the following papers when using this package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">'singscore'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## To cite package 'singscore' in publications use:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Foroutan M, Bhuva D, Lyu R, Horan K, Cursons J, Davis M (2018).</span></span>
<span><span class="co">##   "Single sample scoring of molecular phenotypes." _BMC</span></span>
<span><span class="co">##   bioinformatics_, *19*(1), 404. doi:10.1186/s12859-018-2435-4</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1186/s12859-018-2435-4&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Bhuva D, Cursons J, Davis M (2020). "Stable gene expression for</span></span>
<span><span class="co">##   normalisation and single-sample scoring." _Nucleic Acids Research_,</span></span>
<span><span class="co">##   *48*(19), e113. doi:10.1093/nar/gkaa802</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1093/nar/gkaa802&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span></span>
<span><span class="co">## bibtex=TRUE)', 'toBibtex(.)', or set</span></span>
<span><span class="co">## 'options(citation.bibtex.max=999)'.</span></span></code></pre>
<p><code>singscore</code> implements a simple single-sample gene-set
(gene-signature) scoring method which scores individual samples
independently without relying on other samples in gene expression
datasets. It provides stable scores which are less likely to be affected
by varying sample and gene sizes in datasets and unwanted variations
across samples. The scoring method uses a rank-based statistics and is
quick to compute. For details of the methods please refer to the paper
<span class="citation">(Foroutan et al. 2018)</span>. It also provides
various visualisation functions to further explore results of the
analysis.</p>
<p>Additional packages we have developed can enhance the singscore
workflow:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://www.bioconductor.org/packages/release/data/experiment/html/msigdb.html" class="external-link"><code>msigdb</code></a>
- A package that provides gene-sets from the molecular signatures
database (MSigDB) as a <code>GeneSetCollection</code> object that is
compatible with <code>singscore</code>.</li>
<li>
<a href="https://www.bioconductor.org/packages/release/bioc/html/vissE.html" class="external-link"><code>vissE</code></a>
- A package that can summarise and aid in the interpretation of a list
of significant gene-sets identified by <code>singscore</code> (see <a href="https://davislaboratory.github.io/GenesetAnalysisWorkflow/" class="external-link">tutorial</a>).</li>
<li>
<a href="https://www.bioconductor.org/packages/release/data/experiment/html/emtdata.html" class="external-link"><code>emtdata</code></a>
- The full EMT dataset used in this tutorial (with additional EMT
related datasets).</li>
</ol>
<p>We have also published and made openly available the extensive
tutorials below that demonstrate the variety of ways in which
<code>singscore</code> can be used to gain a better functional
understanding of molecular data:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://f1000research.com/articles/8-776" class="external-link">Using singscore
to predict mutation status in acute myeloid leukemia from transcriptomic
signatures</a>.</li>
<li>
<a href="https://davislaboratory.github.io/GenesetAnalysisWorkflow/" class="external-link">Gene-set
enrichment analysis workshop</a> - available through the <a href="http://app.orchestra.cancerdatasci.org/" class="external-link">Orchestra</a> platform
(search “WEHI Masterclass Day 4: Functional Analysis, single sample gene
set analysis”).</li>
</ol>
</div>
<div class="section level2">
<h2 id="install-singscore-r-package">Install “singscore” R package<a class="anchor" aria-label="anchor" href="#install-singscore-r-package"></a>
</h2>
<p>Install ‘singscore’ from Bioconductor</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"singscore"</span><span class="op">)</span></span></code></pre></div>
<p>The most updated version of ‘singscore’ is hosted on GitHub and can
be easily installed using <code><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">devtools::install_github()</a></code>
function provided by <strong>devtools</strong>, (<a href="https://cran.r-project.org/package=devtools" class="external-link uri">https://cran.r-project.org/package=devtools</a>)</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># You would need to install 'devtools' package first.</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># And install the 'singscore' package from the GitHub repository</span></span>
<span><span class="co"># 'singscore' requires these packages to be installed: methods, stats, graphics, ggplot2, ggsci, grDevices,</span></span>
<span><span class="co">#  ggrepel, plotly, tidyr, plyr, magrittr, reshape, edgeR, RColorBrewer, Biobase, GSEABase, BiocParallel</span></span>
<span> <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'DavisLaboratory/singscore'</span><span class="op">)</span></span>
<span><span class="co"># Set build_vignette = TRUE if would like to browseVignette()</span></span></code></pre></div>
<p>#Scoring samples against a gene-set ## Load datasets</p>
<p>To illustrate the usage of ‘simpleScore()’, we first need to load the
example datasets. The datasets used in this vignette have been built
within the package. You can use the following scripts to load them into
your R environment. Detailed steps of obtaining the full datasets are
included at the end of the vignette. The ‘tgfb_expr_10_se’ dataset was
obtained from [@Foroutanmolcanres.0313.2016] and it is a ten-sample
subset of the original dataset. We are going to score the integrated
TGFb-treated gene expression dataset (4 cases and 6 controls) against a
TGFb gene signature with an up-regulated and down-regulated gene-set
pair [@Foroutanmolcanres.0313.2016].</p>
<p>Gene-sets from the molecular signatures database (MSigDB) can be
accessed via our <code>msigdb</code> R/Bioconductor package (see <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/msigdb/inst/doc/msigdb.html" class="external-link">vignette</a>).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore" class="external-link">singscore</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span><span class="co"># The example expression dataset and gene signatures are included in the package</span></span>
<span><span class="co"># distribution, one can directly access them using the variable names</span></span>
<span></span>
<span><span class="co"># To see the description of 'tgfb_expr_10_se','tgfb_gs_up','tgfb_gs_dn', look at </span></span>
<span><span class="co"># their help pages using:</span></span>
<span></span>
<span><span class="co"># ?tgfb_expr_10_se</span></span>
<span><span class="co"># ?tgfb_gs_up</span></span>
<span><span class="co"># ?tgfb_gs_dn</span></span>
<span></span>
<span><span class="co"># Have a look at the object tgfb_expr_10_se containing gene expression data</span></span>
<span><span class="co"># for 10 samples </span></span>
<span><span class="va">tgfb_expr_10_se</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 11900 10 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(11900): 2 9 ... 729164 752014</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(10): D_Ctrl_R1 D_TGFb_R1 ... Hil_Ctrl_R1 Hil_Ctrl_R2</span></span>
<span><span class="co">## colData names(1): Treatment</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the sample names by</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tgfb_expr_10_se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "D_Ctrl_R1"   "D_TGFb_R1"   "D_Ctrl_R2"   "D_TGFb_R2"   "Hes_Ctrl_R1"</span></span>
<span><span class="co">##  [6] "Hes_TGFb_R1" "Hes_Ctrl_R2" "Hes_TGFb_R2" "Hil_Ctrl_R1" "Hil_Ctrl_R2"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View what tgfb_gs_up/dn contains</span></span>
<span><span class="va">tgfb_gs_up</span></span></code></pre></div>
<pre><code><span><span class="co">## setName: NA </span></span>
<span><span class="co">## geneIds: 19, 87, ..., 402055 (total: 193)</span></span>
<span><span class="co">## geneIdType: Null</span></span>
<span><span class="co">## collectionType: Null </span></span>
<span><span class="co">## details: use 'details(object)'</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tgfb_gs_dn</span></span></code></pre></div>
<pre><code><span><span class="co">## setName: NA </span></span>
<span><span class="co">## geneIds: 136, 220, ..., 161291 (total: 108)</span></span>
<span><span class="co">## geneIdType: Null</span></span>
<span><span class="co">## collectionType: Null </span></span>
<span><span class="co">## details: use 'details(object)'</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the size of the gene sets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">tgfb_gs_up</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 193</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">tgfb_gs_dn</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 108</span></span></code></pre>
<div class="section level3">
<h3 id="sample-scoring">Sample scoring<a class="anchor" aria-label="anchor" href="#sample-scoring"></a>
</h3>
<p>To score samples, the gene expression dataset first needs to be
ranked using the <code><a href="../reference/rankGenes.html">rankGenes()</a></code> function which returns a rank
matrix. This matrix along with the signatures are then passed to the
<code><a href="../reference/simpleScore.html">simpleScore()</a></code> function which returns a data.frame
containing the scores for each sample. When only a single gene-set is
available (i.e. not an up- and down- regulated pair), the same function
can be called by setting the <code>upSet</code> argument to the
gene-set.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The recommended method for dealing with ties in ranking is 'min', you can</span></span>
<span><span class="co"># change by specifying 'tiesMethod' parameter for rankGenes function.</span></span>
<span><span class="va">rankData</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="va">tgfb_expr_10_se</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Given the ranked data and gene signature, simpleScore returns the scores and </span></span>
<span><span class="co"># dispersions for each sample</span></span>
<span><span class="va">scoredf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">rankData</span>, upSet <span class="op">=</span> <span class="va">tgfb_gs_up</span>, downSet <span class="op">=</span> <span class="va">tgfb_gs_dn</span><span class="op">)</span></span>
<span><span class="va">scoredf</span></span></code></pre></div>
<pre><code><span><span class="co">##               TotalScore TotalDispersion    UpScore UpDispersion   DownScore</span></span>
<span><span class="co">## D_Ctrl_R1   -0.088097993        2867.348 0.06096415     3119.390 -0.14906214</span></span>
<span><span class="co">## D_TGFb_R1    0.286994210        2217.970 0.24931565     2352.886  0.03767856</span></span>
<span><span class="co">## D_Ctrl_R2   -0.098964086        2861.418 0.06841242     3129.769 -0.16737650</span></span>
<span><span class="co">## D_TGFb_R2    0.270721958        2378.832 0.25035661     2470.012  0.02036534</span></span>
<span><span class="co">## Hes_Ctrl_R1 -0.002084788        2746.146 0.08046490     3134.216 -0.08254969</span></span>
<span><span class="co">## Hes_TGFb_R1  0.176122839        2597.515 0.22894035     2416.638 -0.05281751</span></span>
<span><span class="co">## Hes_Ctrl_R2  0.016883867        2700.556 0.08817828     3138.664 -0.07129441</span></span>
<span><span class="co">## Hes_TGFb_R2  0.188466953        2455.186 0.23895473     2324.717 -0.05048778</span></span>
<span><span class="co">## Hil_Ctrl_R1 -0.061991164        3039.330 0.08314254     3553.792 -0.14513371</span></span>
<span><span class="co">## Hil_Ctrl_R2 -0.064937366        2959.270 0.07433863     3396.637 -0.13927600</span></span>
<span><span class="co">##             DownDispersion</span></span>
<span><span class="co">## D_Ctrl_R1         2615.306</span></span>
<span><span class="co">## D_TGFb_R1         2083.053</span></span>
<span><span class="co">## D_Ctrl_R2         2593.067</span></span>
<span><span class="co">## D_TGFb_R2         2287.652</span></span>
<span><span class="co">## Hes_Ctrl_R1       2358.075</span></span>
<span><span class="co">## Hes_TGFb_R1       2778.392</span></span>
<span><span class="co">## Hes_Ctrl_R2       2262.448</span></span>
<span><span class="co">## Hes_TGFb_R2       2585.654</span></span>
<span><span class="co">## Hil_Ctrl_R1       2524.868</span></span>
<span><span class="co">## Hil_Ctrl_R2       2521.903</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To view more details of the simpleScore, use ?simpleScore</span></span>
<span><span class="co"># Note that, when only one gene set is available in a gene signature, one can </span></span>
<span><span class="co"># only input values for the upSet argument. In addition, a knownDirection </span></span>
<span><span class="co"># argument can be set to FALSE if the direction of the gene set is unknown.</span></span>
<span></span>
<span><span class="co"># simpleScore(rankData, upSet = tgfb_gs_up, knownDirection = FALSE)</span></span></code></pre></div>
<p>The returned data.frame consists of the scores for the up- and down-
regulated gene-sets along with the combined score
(<code>TotalScore</code>). Dispersion is calculated using the
<code>mad</code> function by default and can be substituted by passing
another function to the <code>dispersionFun</code> argument in
<code><a href="../reference/simpleScore.html">simpleScore()</a></code> such as <code>IQR</code> to calculate the
inter-quartile range.</p>
</div>
<div class="section level3">
<h3 id="sample-scoring-with-a-reduced-number-of-measurements">Sample scoring with a reduced number of measurements<a class="anchor" aria-label="anchor" href="#sample-scoring-with-a-reduced-number-of-measurements"></a>
</h3>
<p>Singscore requires transcriptome-wide measurements whereby all or
most transcripts are measured. This is required to assess the expression
of a gene in relation to the transcriptome. Panel-based transcriptomic
assays measure a much smaller selection of transcripts therefore
assessing the relative expression of each gene becomes a challenge. In
such a setting, stably expressed genes can be used to assess relative
expression. Expression of such genes is invariable across samples
therefore they can be used to calibrate sample-wise gene expression
measurements. This property allows for estimation of transcriptome-wide
ranks with a small set of measurements inclusing a small set of stable
genes. Detailed explanation of the rank estimation procedure using this
logic is available in Bhuva et al. (2020) (manuscript in
preparation).</p>
<p>Stably expressed genes in carcinoma transcriptomes and blood can be
obtained using the <code>getStableGenes</code> function.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#get the top 5 stable genes in carcinomas</span></span>
<span><span class="fu"><a href="../reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span>, type <span class="op">=</span> <span class="st">'carcinoma'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "RBM45"  "BRAP"   "CIAO1"  "TARDBP" "HNRNPK"</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#get the top 5 stable genes in blood</span></span>
<span><span class="fu"><a href="../reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span>, type <span class="op">=</span> <span class="st">'blood'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "RBM45"  "BRAP"   "GOSR1"  "IWS1"   "HNRNPK"</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#get ensembl IDs instead of gene symbold</span></span>
<span><span class="fu"><a href="../reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span>, type <span class="op">=</span> <span class="st">'carcinoma'</span>, id <span class="op">=</span> <span class="st">'ensembl'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSG00000155636" "ENSG00000089234" "ENSG00000144021" "ENSG00000120948"</span></span>
<span><span class="co">## [5] "ENSG00000165119"</span></span></code></pre>
<p>This list can be used to score samples using a variant of singscore
called <em>stingscore</em>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#here we specify a custom set of genes (Entrez IDs)</span></span>
<span><span class="va">stable_genes</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'8315'</span>, <span class="st">'9391'</span>, <span class="st">'23435'</span>, <span class="st">'3190'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#create a dataset with a reduced set of genes (signature genes and stable genes)</span></span>
<span><span class="va">measured</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">stable_genes</span>, <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">tgfb_gs_up</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">tgfb_gs_dn</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">small_tgfb_expr_10</span> <span class="op">&lt;-</span>  <span class="va">tgfb_expr_10_se</span><span class="op">[</span><span class="va">measured</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">small_tgfb_expr_10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 305  10</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#rank genes using stable genes</span></span>
<span><span class="va">rankData_st</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="va">small_tgfb_expr_10</span>, stableGenes <span class="op">=</span> <span class="va">stable_genes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rankData_st</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       D_Ctrl_R1 D_TGFb_R1 D_Ctrl_R2 D_TGFb_R2 Hes_Ctrl_R1 Hes_TGFb_R1</span></span>
<span><span class="co">## 8315        0.2       0.2       0.2       0.2         0.2         0.2</span></span>
<span><span class="co">## 9391        0.4       0.4       0.4       0.4         0.4         0.4</span></span>
<span><span class="co">## 23435       0.6       0.6       0.6       0.6         0.6         0.6</span></span>
<span><span class="co">## 3190        0.8       0.8       0.8       0.8         0.8         0.8</span></span>
<span><span class="co">## 19          0.4       0.4       0.4       0.6         0.4         0.6</span></span>
<span><span class="co">## 87          0.8       1.0       0.8       0.8         0.8         0.8</span></span>
<span><span class="co">##       Hes_Ctrl_R2 Hes_TGFb_R2 Hil_Ctrl_R1 Hil_Ctrl_R2</span></span>
<span><span class="co">## 8315          0.2         0.2         0.2         0.2</span></span>
<span><span class="co">## 9391          0.4         0.4         0.4         0.4</span></span>
<span><span class="co">## 23435         0.6         0.6         0.6         0.6</span></span>
<span><span class="co">## 3190          0.8         0.8         0.8         0.8</span></span>
<span><span class="co">## 19            0.4         0.8         0.4         0.4</span></span>
<span><span class="co">## 87            0.8         0.8         0.8         0.8</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#score samples using stingscore</span></span>
<span><span class="co">#simpleScore invoked with the new rank matrix will execute the stingscore</span></span>
<span><span class="co">#   algorithm</span></span>
<span><span class="va">scoredf_st</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">rankData_st</span>, upSet <span class="op">=</span> <span class="va">tgfb_gs_up</span>, downSet <span class="op">=</span> <span class="va">tgfb_gs_dn</span><span class="op">)</span></span>
<span><span class="va">scoredf_st</span></span></code></pre></div>
<pre><code><span><span class="co">##              TotalScore TotalDispersion     UpScore UpDispersion   DownScore</span></span>
<span><span class="co">## D_Ctrl_R1   -0.04940510         0.14826 -0.10310881      0.29652 0.053703704</span></span>
<span><span class="co">## D_TGFb_R1    0.27645366         0.14826  0.08756477      0.29652 0.188888889</span></span>
<span><span class="co">## D_Ctrl_R2   -0.07259643         0.14826 -0.09481865      0.29652 0.022222222</span></span>
<span><span class="co">## D_TGFb_R2    0.26430628         0.14826  0.08652850      0.29652 0.177777778</span></span>
<span><span class="co">## Hes_Ctrl_R1  0.01732873         0.14826 -0.09378238      0.29652 0.111111111</span></span>
<span><span class="co">## Hes_TGFb_R1  0.18121282         0.29652  0.06269430      0.29652 0.118518519</span></span>
<span><span class="co">## Hes_Ctrl_R2  0.03176933         0.14826 -0.08860104      0.29652 0.120370370</span></span>
<span><span class="co">## Hes_TGFb_R2  0.20667818         0.29652  0.08445596      0.29652 0.122222222</span></span>
<span><span class="co">## Hil_Ctrl_R1 -0.05136250         0.14826 -0.06062176      0.29652 0.009259259</span></span>
<span><span class="co">## Hil_Ctrl_R2 -0.03202840         0.14826 -0.06165803      0.29652 0.029629630</span></span>
<span><span class="co">##             DownDispersion</span></span>
<span><span class="co">## D_Ctrl_R1          0.00000</span></span>
<span><span class="co">## D_TGFb_R1          0.00000</span></span>
<span><span class="co">## D_Ctrl_R2          0.00000</span></span>
<span><span class="co">## D_TGFb_R2          0.00000</span></span>
<span><span class="co">## Hes_Ctrl_R1        0.00000</span></span>
<span><span class="co">## Hes_TGFb_R1        0.29652</span></span>
<span><span class="co">## Hes_Ctrl_R2        0.00000</span></span>
<span><span class="co">## Hes_TGFb_R2        0.29652</span></span>
<span><span class="co">## Hil_Ctrl_R1        0.00000</span></span>
<span><span class="co">## Hil_Ctrl_R2        0.00000</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#plot the two scores against each other</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">scoredf</span><span class="op">$</span><span class="va">TotalScore</span>, <span class="va">scoredf_st</span><span class="op">$</span><span class="va">TotalScore</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="singscore_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p>Scores computed using the classic singscore and the modified
stingscore will be similar (correlated if not equal) if the selection of
stable genes is good. Ideal stable genes will have stable gene
expression across the samples being investigated and should cover a wide
range of the expression spectrum. The genes produced by
<code>getStableGenes</code> fulfill these conditions. Our approach using
the product of ranks can be used to determine a set of stable genes for
a new context (manuscript in preparation).</p>
</div>
</div>
<div class="section level2">
<h2 id="visualisation-and-diagnostic-functions">Visualisation and diagnostic functions<a class="anchor" aria-label="anchor" href="#visualisation-and-diagnostic-functions"></a>
</h2>
<p>In this section, we show example usages of the visualisation
functions included in this package.</p>
<div class="section level3">
<h3 id="plot-rank-densities">Plot rank densities<a class="anchor" aria-label="anchor" href="#plot-rank-densities"></a>
</h3>
<p>Scores of each sample are stored in <code>scoredf</code>. We can use
the <code>plotRankDensity</code> function to plot the ranks of genes in
the gene-sets for a specific sample. We plot the rank distribution for
the second sample in <code>rankData</code> which combines a density plot
(densities calculated using KDE) with a barcode plot. Please note that
since we are subsetting the data.frame rankData by one column, we set
<code>drop = FALSE</code> to maintain the structure of the
data.frame/matrix.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#  You can provide the upSet alone when working with unpaired gene-sets </span></span>
<span><span class="co"># We plot the second sample in rankData, view it by </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rankData</span><span class="op">[</span>,<span class="fl">2</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    D_TGFb_R1</span></span>
<span><span class="co">## 2       1255</span></span>
<span><span class="co">## 9       7611</span></span>
<span><span class="co">## 10      1599</span></span>
<span><span class="co">## 12      3682</span></span>
<span><span class="co">## 13      3599</span></span>
<span><span class="co">## 14     10013</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotRankDensity.html">plotRankDensity</a></span><span class="op">(</span><span class="va">rankData</span><span class="op">[</span>,<span class="fl">2</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, upSet <span class="op">=</span> <span class="va">tgfb_gs_up</span>, </span>
<span>                downSet <span class="op">=</span> <span class="va">tgfb_gs_dn</span>, isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: The dot-dot notation (`..density..`) was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use `after_stat(density)` instead.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">singscore</span> package.</span></span>
<span><span class="co">##   Please report the issue at</span></span>
<span><span class="co">##   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/DavisLaboratory/singscore/issues&gt;</span>.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_segment(aes(y = ymap[upDown], xend = Ranks, yend =</span></span>
<span><span class="co">## yendmap[upDown], : Ignoring unknown aesthetics: <span style="color: #00BB00;">text</span></span></span></code></pre>
<p><img src="singscore_files/figure-html/plotdt-1.png" width="768"></p>
<p>Setting <code>isInteractive = TRUE</code> generates an interactive
plot using the <code>plotly</code> package. Hovering over the bars in
the interactive plot allows you to get information such as the
normalised rank (between 0 and 1) and ID of the gene represented by the
bar. For the rest of the plotting functions, the
<code>isInteractive = TRUE</code> argument has the same behavior.</p>
</div>
<div class="section level3">
<h3 id="plot-dispersions-of-scores">Plot dispersions of scores<a class="anchor" aria-label="anchor" href="#plot-dispersions-of-scores"></a>
</h3>
<p>Function <code>plotDispersion</code> generates the scatter plots of
the ‘score VS. dispersions’ for the total scores, the up scores and the
down score of samples. It requires the scored data.frame from
<code>simpleScore</code> function and annotations (via
<code>annot</code> parameter) can be used for coloring the points.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#  Get the annotations of samples by their sample names</span></span>
<span><span class="va">tgfbAnnot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>SampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tgfb_expr_10_se</span><span class="op">)</span>,</span>
<span>                       Type <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="va">tgfbAnnot</span><span class="op">$</span><span class="va">Type</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Ctrl"</span>, <span class="va">tgfbAnnot</span><span class="op">$</span><span class="va">SampleID</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="st">"Control"</span></span>
<span><span class="va">tgfbAnnot</span><span class="op">$</span><span class="va">Type</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"TGFb"</span>, <span class="va">tgfbAnnot</span><span class="op">$</span><span class="va">SampleID</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="st">"TGFb"</span></span>
<span></span>
<span><span class="co"># Sample annotations</span></span>
<span><span class="va">tgfbAnnot</span><span class="op">$</span><span class="va">Type</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "Control" "TGFb"    "Control" "TGFb"    "Control" "TGFb"    "Control"</span></span>
<span><span class="co">##  [8] "TGFb"    "Control" "Control"</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDispersion.html">plotDispersion</a></span><span class="op">(</span><span class="va">scoredf</span>,annot <span class="op">=</span> <span class="va">tgfbAnnot</span><span class="op">$</span><span class="va">Type</span>,isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="singscore_files/figure-html/plotds-1.png" width="768"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To see an interactive version powered by 'plotly', simply set the </span></span>
<span><span class="co"># 'isInteractive' = TRUE, i.e :</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># plotDispersion(scoredf,annot = tgfbAnnot$Type,isInteractive = TRUE)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-score-landscape">Plot score landscape<a class="anchor" aria-label="anchor" href="#plot-score-landscape"></a>
</h3>
<p><code>plotScoreLandscape</code> plots the scores of the samples
against two different gene signatures in a landscape for exploring their
relationships.</p>
<p>There are two styles of the landscape plot (i.e scatter and hexBin
plot). When the number of samples in the gene expression dataset is
above the default threshold (100), <code>plotScoreLandscape</code>
generates a hex bin plot otherwise a scatter plot. The threshold can be
modified by changing the <code>hexMin</code>.</p>
<p>In order to better demonstrate the usage of
<code>plotScoreLandscape</code>, we load some additional datasets
consisting of pre-computed scores of larger public datasets.
<code>scoredf_ccle_epi</code> and <code>scoredf_ccle_mes</code> are two
scored results of a CCLE dataset <span class="citation">(Barretina et
al. 2012)</span> against an epithelial gene signature and mesenchymal
gene signature <span class="citation">(Tan et al. 2014)</span>
respectively. For details on how to obtain the dataset please see the
section at the end of the vignette.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotScoreLandscape.html">plotScoreLandscape</a></span><span class="op">(</span><span class="va">scoredf_ccle_epi</span>, <span class="va">scoredf_ccle_mes</span>, </span>
<span>                   scorenames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ccle-EPI'</span>,<span class="st">'ccle-MES'</span><span class="op">)</span>,hexMin <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="singscore_files/figure-html/loadCCLE-1.png" width="768"></p>
<p>Similarly, pre-computed scores for the <a href="https://cancergenome.nih.gov/" class="external-link">TCGA</a> breast cancer RNA-seq
dataset against epithelial and mesenchymal gene signatures are stored in
<code>scoredf_tcga_epi</code> and <code>scoredf_tcga_mes</code>
respectively <span class="citation">(Tan et al. 2014)</span>. The
utility of this function is enhanced when the number of samples is
large.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tcgaLandscape</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotScoreLandscape.html">plotScoreLandscape</a></span><span class="op">(</span><span class="va">scoredf_tcga_epi</span>, <span class="va">scoredf_tcga_mes</span>, </span>
<span>                   scorenames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'tcga_EPI'</span>,<span class="st">'tcga_MES'</span><span class="op">)</span>, isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tcgaLandscape</span></span></code></pre></div>
<p><img src="singscore_files/figure-html/loadTCGA-1.png" width="768"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To get an interactive version of plot, set isInteractive = TRUE</span></span></code></pre></div>
<p>You can also project new data points onto the landscape plot
generated above by using the <code>projectScoreLandscape</code>
function. For example, the plot below overlays 3 CCLE samples onto the
TCGA epithelial-mesenchymal landscape. Points are labeled with their
sample names by default.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Project on the above generated 'tcgaLandscape' plot</span></span>
<span><span class="fu"><a href="../reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>plotObj <span class="op">=</span> <span class="va">tcgaLandscape</span>, <span class="va">scoredf_ccle_epi</span>, </span>
<span>                      <span class="va">scoredf_ccle_mes</span>,</span>
<span>                      subSamples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scoredf_ccle_epi</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                      annot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scoredf_ccle_epi</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                      sampleLabels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                      isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in geom_point(aes(text = SampleText, colour = Class), shape = 21, :</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">text</span></span></span></code></pre>
<p><img src="singscore_files/figure-html/projectScore-1.png" width="768"></p>
<p>Custom labels can be provided by passing a character vector to the
<code>sampleLabels</code> argument.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>plotObj <span class="op">=</span> <span class="va">tcgaLandscape</span>, <span class="va">scoredf_ccle_epi</span>, <span class="va">scoredf_ccle_mes</span>,</span>
<span>                      subSamples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scoredf_ccle_epi</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span>,<span class="fl">5</span>,<span class="fl">8</span>,<span class="fl">9</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                      sampleLabels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'l1'</span>,<span class="st">'l2'</span>,<span class="st">'l3'</span>,<span class="st">'l4'</span>,<span class="st">'l5'</span><span class="op">)</span>,</span>
<span>                      annot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scoredf_ccle_epi</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span>,<span class="fl">5</span>,<span class="fl">8</span>,<span class="fl">9</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                      isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in geom_point(aes(text = SampleText, colour = Class), shape = 21, :</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">text</span></span></span></code></pre>
<p><img src="singscore_files/figure-html/projectScore2-1.png" width="768"></p>
</div>
</div>
<div class="section level2">
<h2 id="estimate-empirical-p-values-for-the-obtained-scores-in-individual-samples-and-plot-null-distributions">Estimate empirical p-values for the obtained scores in individual
samples and plot null distributions<a class="anchor" aria-label="anchor" href="#estimate-empirical-p-values-for-the-obtained-scores-in-individual-samples-and-plot-null-distributions"></a>
</h2>
<div class="section level3">
<h3 id="permutation-test">Permutation test<a class="anchor" aria-label="anchor" href="#permutation-test"></a>
</h3>
<p>Hypothesis testing of the calculated scores is performed using a
permutation test. The null hypothesis is that the gene-set is not
enriched in the sample. For each sample, gene labels are randomly
shuffled and scores computed against the gene-set. This is done <span class="math inline">\(B\)</span> times to generate the null
distribution. The <code><a href="../reference/generateNull.html">generateNull()</a></code> function computes these
for multiple samples (<span class="math inline">\(n\)</span>)
simultaneously resulting in an <span class="math inline">\(n \times
B\)</span> matrix with permuted scores along the columns for each
sample.</p>
<p>The permutation function has parallel computing features provided by
using <a href="http://bioconductor.org/packages/release/bioc/html/BiocParallel.html" class="external-link"><code>BiocParallel</code></a></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># tgfb_gs_up : up regulated gene set in the tgfb gene signature</span></span>
<span><span class="co"># tgfb_gs_dn : down regulated gene set in the tgfb gene signature</span></span>
<span></span>
<span><span class="co"># This permutation function uses BiocParallel::bplapply() parallel function, by </span></span>
<span><span class="co"># supplying the first 5 columns of rankData, we generate Null distribution for </span></span>
<span><span class="co"># the first 5 samples.</span></span>
<span></span>
<span><span class="co"># detect how many CPU cores are available for your machine</span></span>
<span><span class="co"># parallel::detectCores()</span></span>
<span></span>
<span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="co"># Provide the generateNull() function the number of cores you would like</span></span>
<span><span class="co"># the function to use by passing the ncore parameter</span></span>
<span></span>
<span><span class="va">permuteResult</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/generateNull.html">generateNull</a></span><span class="op">(</span></span>
<span>    upSet <span class="op">=</span> <span class="va">tgfb_gs_up</span>,</span>
<span>    downSet <span class="op">=</span> <span class="va">tgfb_gs_dn</span>,</span>
<span>    rankData <span class="op">=</span> <span class="va">rankData</span>,</span>
<span>    subSamples <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,</span>
<span>    centerScore <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    knownDirection <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    B <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>    ncores <span class="op">=</span> <span class="va">ncores</span>,</span>
<span>    seed <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    useBPPARAM <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co"># Note here, the useBPPARAM parameter allows user to supply a BPPARAM variable </span></span>
<span><span class="co"># as a parameter which decides the type of parallel ends to use.</span></span>
<span><span class="co"># such as </span></span>
<span><span class="co"># snow &lt;-  BiocParallel::SnowParam(type = "SOCK")</span></span>
<span><span class="co"># permuteResult &lt;-  generateNull(upSet = tgfb_gs_up, downSet = tgfb_gs_dn,</span></span>
<span><span class="co"># rankData[,1:5],  B = 1000, seed = 1,ncores = ncores, useBPPARAM = snow)</span></span>
<span></span>
<span><span class="co"># If you are not sure about this, just leave the value as NULL and set how many</span></span>
<span><span class="co"># CPU cores to use via the ncores argument. It will use the default parallel </span></span>
<span><span class="co"># backend for your machine.</span></span>
<span></span>
<span><span class="co"># For more information, please see the help page for ?generateNull()</span></span>
<span><span class="co"># Please also note that one should pass the same values to the upSet, </span></span>
<span><span class="co"># downSet, centerScore and knownDirection arguments as what they provide </span></span>
<span><span class="co"># for the simpleScore() function to generate a proper null distribution.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">permuteResult</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     D_Ctrl_R1   D_TGFb_R1   D_Ctrl_R2   D_TGFb_R2 Hes_Ctrl_R1</span></span>
<span><span class="co">## 1 -0.03108049 -0.02355802 -0.03296665 -0.02552833 -0.02976792</span></span>
<span><span class="co">## 2  0.05924161  0.05508062  0.05900367  0.05473372  0.05781921</span></span>
<span><span class="co">## 3 -0.06808434 -0.06405468 -0.07658483 -0.06884367 -0.06999305</span></span>
<span><span class="co">## 4  0.06030101  0.08181637  0.05814608  0.08205247  0.07089498</span></span>
<span><span class="co">## 5 -0.05495649 -0.04898792 -0.06288874 -0.05624015 -0.05125311</span></span>
<span><span class="co">## 6  0.04852306  0.03894352  0.05051753  0.03736856  0.04133476</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="estimate-empirical-p-values">Estimate empirical p-values<a class="anchor" aria-label="anchor" href="#estimate-empirical-p-values"></a>
</h3>
<p><span class="math inline">\(p\)</span>-values can be estimated using
the <code><a href="../reference/getPvals.html">getPvals()</a></code> function by providing the null distributions
calculated above. Unless all permutations are exhausted (mostly
infeasible), the minimum <span class="math inline">\(p\)</span>-value
obtained is <span class="math inline">\(\frac{1}{B}\)</span>.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPvals.html">getPvals</a></span><span class="op">(</span><span class="va">permuteResult</span>, <span class="va">scoredf</span>, subSamples <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># getPval returns p-values for each individual sample.</span></span>
<span><span class="co"># show the p-values for first 5 samples</span></span>
<span><span class="va">pvals</span></span></code></pre></div>
<pre><code><span><span class="co">##   D_Ctrl_R1   D_TGFb_R1   D_Ctrl_R2   D_TGFb_R2 Hes_Ctrl_R1 </span></span>
<span><span class="co">##       0.996       0.001       0.999       0.001       0.530</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="plot-null-distribution">Plot null distribution<a class="anchor" aria-label="anchor" href="#plot-null-distribution"></a>
</h3>
<p>Plot the null distributions for the first sample with the estimated
<span class="math inline">\(p\)</span>-value labelled. The function uses
the <code>sampleNames</code> parameter to decide which samples to
plot.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotNull.html">plotNull</a></span><span class="op">(</span><span class="va">permuteResult</span>, <span class="va">scoredf</span>, <span class="va">pvals</span>, sampleNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="singscore_files/figure-html/plotNull1-1.png" width="768"></p>
<p>You can provide multiple sample names to plot these samples in one
plot. For example, plot the first 2 samples.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the null distributions for the first 2 samples and the p-values</span></span>
<span><span class="co"># We can see from the plot, the control samples are not significant and TGFb </span></span>
<span><span class="co"># samples are very significant with very low p-values</span></span>
<span><span class="fu"><a href="../reference/plotNull.html">plotNull</a></span><span class="op">(</span><span class="va">permuteResult</span>, <span class="va">scoredf</span>, <span class="va">pvals</span>, sampleNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using  as id variables</span></span></code></pre>
<p><img src="singscore_files/figure-html/plotNull2-1.png" width="1152"></p>
</div>
</div>
<div class="section level2">
<h2 id="more-on-the-datasets">More on the datasets<a class="anchor" aria-label="anchor" href="#more-on-the-datasets"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<p>TGFb-EMT data</p>
<p>In the examples above, we loaded a gene expression matrix data
<code>tgfb_expr_10_se</code>. This dataset is a ten-sample subset of a
full dataset originally from the integrated TGFb-EMT data published by
[@Foroutanmolcanres.0313.2016]. The full dataset can be accessed here <a href="https://figshare.com/articles/comBat_corrected_Foroutan_et_al_2017/5682862" class="external-link uri">https://figshare.com/articles/comBat_corrected_Foroutan_et_al_2017/5682862</a>.
The <code>tgfb_gs_up</code> and <code>tgfb_gs_dn</code> are the derived
up-regulated/down-regulated genes of the TGFb-induced EMT gene signature
by [@Foroutanmolcanres.0313.2016] (see Table S1. TGFβ-EMT
signature).</p>
</li>
<li><p>CCLE dataset</p></li>
</ol>
<p><code>scoredf_ccle_epi</code> and <code>scoredf_ccle_mes</code> are
two data frames of the pre-computed scores using The Cancer Cell Line
Encyclopaedia (CCLE) breast cancer cell line RNA-seq dataset <span class="citation">(Barretina et al. 2012)</span>. The CCLE dataset was
normalised by TPM and can be downloaded from <a href="https://www.synapse.org/#!Synapse:syn5612998" class="external-link uri">https://www.synapse.org/#!Synapse:syn5612998</a> Cell lines
were scored against the epithelial and mesenchymal gene signatures,
which were obtained from <span class="citation">(Tan et al. 2014)</span>
and can be found in the ‘Table S1B. Generic EMT signature for cell line’
in the supplementary files.</p>
<ol start="3" style="list-style-type: decimal">
<li>
<p>TCGA cancer samples dataset</p>
<p><code>scoredf_tcga_epi</code> and <code>scoredf_tcga_mes</code> are
two data frames of the pre-computed scores using The Cancer Genome Atlas
(TCGA) breast cancer RNA-seq data (RSEM normalised) <span class="citation">(The Cancer Genome Atlas Network 2012)</span> against
the epithelial and mesenchymal gene signatures respectively. The gene
signatures were obtained from <span class="citation">(Tan et al.
2014)</span> and can be found in the ‘Table S1A. Generic EMT signature
for tumour’. The TCGA dataset was downloaded from The UCSC Cancer
Genomics Browser in February 2016 (<a href="https://genome-cancer.ucsc.edu" class="external-link uri">https://genome-cancer.ucsc.edu</a>)</p>
</li>
</ol>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># preview the scored CCLE samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scoredf_ccle_epi</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           TotalScore TotalDispersion</span></span>
<span><span class="co">## HCC2157   0.22060031        5926.693</span></span>
<span><span class="co">## AU565     0.30387843        3884.412</span></span>
<span><span class="co">## UACC.893  0.30615947        3527.847</span></span>
<span><span class="co">## BT.483    0.30493150        4158.693</span></span>
<span><span class="co">## Hs.739.T -0.06886754        4281.007</span></span>
<span><span class="co">## HCC2218   0.31230039        3331.402</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># preview the scored TCGA samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scoredf_tcga_epi</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 TotalScore TotalDispersion</span></span>
<span><span class="co">## TCGA.3C.AAAU.01  0.2722128        3952.612</span></span>
<span><span class="co">## TCGA.3C.AALI.01  0.3209824        2835.472</span></span>
<span><span class="co">## TCGA.3C.AALJ.01  0.2531933        4174.260</span></span>
<span><span class="co">## TCGA.3C.AALK.01  0.3042650        3234.292</span></span>
<span><span class="co">## TCGA.4H.AAAK.01  0.2682847        4222.445</span></span>
<span><span class="co">## TCGA.5L.AAT0.01  0.2560761        4132.747</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] GSEABase_1.62.0      graph_1.78.0         annotate_1.78.0     </span></span>
<span><span class="co">##  [4] XML_3.99-0.14        AnnotationDbi_1.62.1 IRanges_2.34.1      </span></span>
<span><span class="co">##  [7] S4Vectors_0.38.1     Biobase_2.60.0       BiocGenerics_0.46.0 </span></span>
<span><span class="co">## [10] singscore_1.19.1     BiocStyle_2.28.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] DBI_1.1.3                   bitops_1.0-7               </span></span>
<span><span class="co">##  [3] rlang_1.1.1                 magrittr_2.0.3             </span></span>
<span><span class="co">##  [5] matrixStats_1.0.0           compiler_4.3.1             </span></span>
<span><span class="co">##  [7] RSQLite_2.3.1               png_0.1-8                  </span></span>
<span><span class="co">##  [9] systemfonts_1.0.4           vctrs_0.6.3                </span></span>
<span><span class="co">## [11] reshape2_1.4.4              stringr_1.5.0              </span></span>
<span><span class="co">## [13] pkgconfig_2.0.3             crayon_1.5.2               </span></span>
<span><span class="co">## [15] fastmap_1.1.1               XVector_0.40.0             </span></span>
<span><span class="co">## [17] labeling_0.4.2              utf8_1.2.3                 </span></span>
<span><span class="co">## [19] rmarkdown_2.22              ragg_1.2.5                 </span></span>
<span><span class="co">## [21] purrr_1.0.1                 bit_4.0.5                  </span></span>
<span><span class="co">## [23] xfun_0.39                   zlibbioc_1.46.0            </span></span>
<span><span class="co">## [25] cachem_1.0.8                GenomeInfoDb_1.36.1        </span></span>
<span><span class="co">## [27] jsonlite_1.8.5              blob_1.2.4                 </span></span>
<span><span class="co">## [29] highr_0.10                  DelayedArray_0.26.3        </span></span>
<span><span class="co">## [31] reshape_0.8.9               BiocParallel_1.34.2        </span></span>
<span><span class="co">## [33] parallel_4.3.1              R6_2.5.1                   </span></span>
<span><span class="co">## [35] bslib_0.5.0                 stringi_1.7.12             </span></span>
<span><span class="co">## [37] RColorBrewer_1.1-3          limma_3.56.2               </span></span>
<span><span class="co">## [39] GenomicRanges_1.52.0        jquerylib_0.1.4            </span></span>
<span><span class="co">## [41] Rcpp_1.0.10                 bookdown_0.34              </span></span>
<span><span class="co">## [43] SummarizedExperiment_1.30.2 knitr_1.43                 </span></span>
<span><span class="co">## [45] Matrix_1.5-4.1              tidyselect_1.2.0           </span></span>
<span><span class="co">## [47] yaml_2.3.7                  codetools_0.2-19           </span></span>
<span><span class="co">## [49] lattice_0.21-8              tibble_3.2.1               </span></span>
<span><span class="co">## [51] plyr_1.8.8                  withr_2.5.0                </span></span>
<span><span class="co">## [53] KEGGREST_1.40.0             evaluate_0.21              </span></span>
<span><span class="co">## [55] desc_1.4.2                  Biostrings_2.68.1          </span></span>
<span><span class="co">## [57] pillar_1.9.0                BiocManager_1.30.21        </span></span>
<span><span class="co">## [59] MatrixGenerics_1.12.2       generics_0.1.3             </span></span>
<span><span class="co">## [61] rprojroot_2.0.3             RCurl_1.98-1.12            </span></span>
<span><span class="co">## [63] ggplot2_3.4.2               munsell_0.5.0              </span></span>
<span><span class="co">## [65] scales_1.2.1                xtable_1.8-4               </span></span>
<span><span class="co">## [67] glue_1.6.2                  tools_4.3.1                </span></span>
<span><span class="co">## [69] hexbin_1.28.3               locfit_1.5-9.8             </span></span>
<span><span class="co">## [71] fs_1.6.2                    grid_4.3.1                 </span></span>
<span><span class="co">## [73] tidyr_1.3.0                 edgeR_3.42.4               </span></span>
<span><span class="co">## [75] colorspace_2.1-0            GenomeInfoDbData_1.2.10    </span></span>
<span><span class="co">## [77] cli_3.6.1                   textshaping_0.3.6          </span></span>
<span><span class="co">## [79] fansi_1.0.4                 S4Arrays_1.0.4             </span></span>
<span><span class="co">## [81] dplyr_1.1.2                 gtable_0.3.3               </span></span>
<span><span class="co">## [83] sass_0.4.6                  digest_0.6.32              </span></span>
<span><span class="co">## [85] ggrepel_0.9.3               farver_2.1.1               </span></span>
<span><span class="co">## [87] memoise_2.0.1               htmltools_0.5.5            </span></span>
<span><span class="co">## [89] pkgdown_2.0.7.9000          lifecycle_1.0.3            </span></span>
<span><span class="co">## [91] httr_1.4.6                  bit64_4.0.5</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-barretina2012cancer" class="csl-entry">
Barretina, Jordi, Giordano Caponigro, Nicolas Stransky, Kavitha
Venkatesan, Adam A Margolin, Sungjoon Kim, Christopher J Wilson, et al.
2012. <span>“The Cancer Cell Line Encyclopedia Enables Predictive
Modelling of Anticancer Drug Sensitivity.”</span> <em>Nature</em> 483
(7391): 603–7.
</div>
<div id="ref-Foroutan231217" class="csl-entry">
Foroutan, Momeneh, Dharmesh D Bhuva, Ruqian Lyu, Kristy Horan, Joseph
Cursons, and Melissa J Davis. 2018. <span>“Single Sample Scoring of
Molecular Phenotypes.”</span> <em>BMC Bioinformatics</em> 19 (1): 404.
<a href="https://doi.org/10.1186/s12859-018-2435-4" class="external-link">https://doi.org/10.1186/s12859-018-2435-4</a>.
</div>
<div id="ref-Ep-MES" class="csl-entry">
Tan, Tuan Zea, Qing Hao Miow, Yoshio Miki, Tetsuo Noda, Seiichi Mori,
Ruby Yun-Ju Huang, and Jean Paul Thiery. 2014.
<span>“Epithelial-Mesenchymal Transition Spectrum Quantification and Its
Efficacy in Deciphering Survival and Drug Responses of Cancer
Patients.”</span> <em>EMBO Molecular Medicine</em> 6 (10): 1279–93. <a href="https://doi.org/10.15252/emmm.201404208" class="external-link">https://doi.org/10.15252/emmm.201404208</a>.
</div>
<div id="ref-TCGAcite" class="csl-entry">
The Cancer Genome Atlas Network. 2012. <span>“Comprehensive Molecular
Portraits of Human Breast Tumors.”</span> <em>Nature</em> 490 (7418):
61–70. <a href="https://doi.org/10.1038/nature11412" class="external-link">https://doi.org/10.1038/nature11412</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Dharmesh D. Bhuva, Ruqian Lyu, Momeneh Foroutan, Malvika Kharbanda.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
